### PR TITLE
ci(e2e): Refactor screenrecorder and framebuffer logs

### PR DESCRIPTION
**Pull request by scinos**
#### Changes proposed in this Pull Request

* Split `video-recorder.js` into two set of hooks: framebuffer and video recorder, as they are separate concerns.
* Drop support for `neverSaveScreenshots` and `saveAllScreenshots`. They are not documented anywhere, and from what I can see, never used.
* Drop support for adding the URL as a png tag. You couldn't easily read them back without specialized tools anyway.
* Clean up the logic to toggle framebuffer and video recording support. Now it is centralized in one single file instead of every method checking if video was available.
* Drop slack notifications when a screenshot couldn't be saved.
* Save videos and screenshots in the same directory for easier asset management
* Add hook to close the browser. Root Hooks run after the hooks declared in `./mocha-hooks.js`. Therefore, closing the browser must be the last `afterAll` hook in Root Hooks, in case previous `afterAll` hooks want to use it (eg: to take a screenshot)

#### Testing instructions

Verify e2e build still saves screenshots and video recordings.

Example:
image

### CODE REVIEW

File: test/e2e/lib/hooks/index.js [hunk 1]
```diff
@@ -1,10 +1,50 @@
...
+		afterEach: [],
+	};
+
+	if ( isVideoEnabled() ) {
+		const displayNum = await getFreeDisplay();
+
+		const { startFramebuffer, stopFramebuffer, takeScreenshot } = buildFramebufferHooks(
+			displayNum
+		);
+		const { startVideoRecording, saveVideoRecording, stopVideoRecording } = buildVideoHooks(
```
Comment (worldomonation) [inline test/e2e/lib/hooks/index.js:36]: I haven't seen this syntax (for root hooks) before. Do you have somewhere I can read up on this?

**Review by worldomonation — APPROVED** [review]
Was able to verify that video and screenshot artifacts are generated for failed cases.

Looks like a safe change and I can't identify any code style issues either.

File: test/e2e/lib/hooks/index.js (hunk 1 above)
Comment (scinos) [reply test/e2e/lib/hooks/index.js:36]: It's called [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring)

Comment (worldomonation) [reply test/e2e/lib/hooks/index.js:36]: Sorry, I meant moreso in the sense of how you were building the hooks. I hadn’t seen hooks built this way in documentation.

Comment (scinos) [reply test/e2e/lib/hooks/index.js:36]: Ah I see what you mean. I don't know if this pattern has a name (edit: I guess its a variant of the [Revealing Module pattern](https://gist.github.com/zcaceres/bb0eec99c02dda6aac0e041d0d4d7bf2))

It's worth mentioning that at this point, those functions are not hooks, they haven't been attached to Mocha yet. So this is a function `buildFramebufferHooks` that returns an object with three functions `{ startFramebuffer, stopFramebuffer, takeScreenshot }`.

I'm using it so those three functions can have access to the same `displayNum` (defined here) and `xvfb` (defined in `framebuffer.js`), without exposing those values as globals.

**Pull request merged**